### PR TITLE
[RW-4135][risk=no] Add non null constraint for user.contact_email

### PR DESCRIPTION
**Pull request by ericsong**
I've already run the necessary SQL updates across the environments so that every user has a contact_email.

---
**PR checklist**

- [ ] This PR meets the Acceptance Criteria in the JIRA story
- [ ] The JIRA story has been moved to Dev Review
- [ ] This PR includes appropriate unit tests
- [ ] I have run and tested this change locally
- [ ] If this includes an API change, I have updated the appropriate Swagger definitions and notified API consumers
- [ ] If this includes a new feature flag, I have created and linked new JIRA tickets to (a) turn on the feature flag and (b) remove it later

### CODE REVIEW

**Comment by jaycarlton** [comment]
🎖

File: api/src/main/java/org/pmiops/workbench/db/dao/UserServiceImpl.java [hunk 1]
```diff
@@ -307,7 +307,7 @@ public DbUser createUser(final Userinfoplus oAuth2Userinfo) {
         oAuth2Userinfo.getGivenName(),
         oAuth2Userinfo.getFamilyName(),
         oAuth2Userinfo.getEmail(),
-        null,
+        oAuth2Userinfo.getEmail(),
```
Comment (ericsong) [inline api/src/main/java/org/pmiops/workbench/db/dao/UserServiceImpl.java]: This covers the edge case where the user already exists in Gsuite when accessing the workbench for the first time. 

In this case, I decided to just set it to the workbench username email (what we did for the backfill).

It's not the best solution since this email is non reachable but this is an edge case that should only occur in test/local environments.
